src/Devices/GateReader/GateReaderHost.cs

```diff
@@ -4,28 +4,30 @@
 
 namespace Transportation.Demo.Devices.GateReader
 {
-    class TransportationDeviceGateReaderClient
+    class GateReaderhost
     {
-        public static TransportationDeviceClient deviceClient;
-        public static string connectionString;
+        public static GateReaderDevice myGateReader;
 
         static void Main(string[] args)
         {
             Console.WriteLine("Transportation Demo- Simulated Gate/Reader. Ctrl-C to exit.\n");
 
-            // Connect to the IoT hub using the MQTT protocol
-            connectionString = ConfigurationHandler.getConfig("AppSettings", "IoTConnectionString");
-            deviceClient = new TransportationDeviceClient(connectionString);
+            // create our simulated device
+            myGateReader = new GateReaderDevice(ConfigurationHandler.getConfig("AppSettings", "IoTConnectionString"));
+            // start the device running
+            myGateReader.StartAllEvents(); 
 
+            // put the host into a loop
             while (true)
             {
                 //do stuff
-
                 Thread.Sleep(30000);
```

Comment:
I don't get it. How will we be able to break this loop? To call `StopAllEvents`

Reply:
It doesn't really. When you launch the program is states to press ctrl-c to end the program which terminates is, essentially breaking it out of the loop. The "stop is just there for demonstrative purposes. :)